spec/spec.tex

```diff
@@ -958,6 +958,99 @@ \subsubsection{Write Under Write Behaviour}
 
 In all cases, if a memory location is written to by more than one port on the same cycle, the stored value is undefined.
 
+\subsubsection{Inferred Memory Ports}
+
+An inferred memory port is one that is not explicitly declared in the original definition of the memory.
+Extra memory ports can be inferred for any memory that is in scope with the `read mport` and `write mport` statements.
+For this section, we will consider the following memory with no explicitly-declared ports.
+
+\begin{lstlisting}
+mem syncmem :
+  data-type => UInt<32>
+  depth => 256
+  read-latency => 1
+  write-latency => 1
+  read-under-write => undefined
+\end{lstlisting}
+
+\subsubsubsection{Inferred Read Ports}
+
+An inferred read port statement has four mandatory arguments: the name of the
+port, the memory to access, the address signal, and the clock. This has the
+effect of implicitly declaring the specified port and connecting the address and
+clock signals to the corresponding fields (`addr` and `clk`) of the new port.
+Thereafter, references to the port will implicitly yield the read data.
+
+Inferred read ports may also accept an optional enable signal, which is
+implicity connected to the `en` field of the new read port. If this argument is
+omitted, the read occurs unconditionally, regardless of the conditional context
+that contains the inferred read port.
+
+In this example, the read data from the inferred read port `r` is
+unconditionally accessible by referring to the signal `r`, even though the
+result is conditionally assigned to the signal `bus.rdata`.
+
+\begin{lstlisting}
+read mport r = syncmem[cmd.addr], clk ; unconditional read
+when (cmd.ren) :
+  resp.rdata <= r
+\end{lstlisting}
+
+In contrast, when the enable signal is provided explicitly, the memory read
+operation itself is conditional on the enable signal.
+
+\begin{lstlisting}
+read mport r = syncmem[cmd.addr], clk, cmd.ren ; conditional read
+when (cmd.ren) :
+  resp.rdata <= r
+\end{lstlisting}
+
```

Comment:
I think we also need an example where a read port is declared inside a when scope, in order to clarify what happens in that case.